config.py

```diff
@@ -6,26 +23,87 @@
 from flask import Flask
 
 
+# Load dotenv early so that `_env` will work in the class definitions below.
 load_dotenv()
 
+#: The test environment. For unit tests only.
+TESTING = "testing"
+#: The development environment. For local development.
+DEVELOPMENT = "development"
+#: The production environment. For production serving.
+PRODUCTION = "production"
+
 
 def _make_path(path: Path):
     path.mkdir(parents=True, exist_ok=True)
     return path
 
 
-def _env(key: str, default=None) -> str:
+def _env(key: str, default=None) -> Optional[str]:
     """Fetch a value from the local environment.
 
     :param key: the envvar to fetch
     """
     return os.environ.get(key, default)
 
 
-class UnitTestConfig:
+class BaseConfig:
+    """Base settings for all configs."""
+
+    # Core settings
+    # -------------
+
+    #: The Flask app environment ("production", "development", etc.). We set
+    #: this explicitly so that Celery can have access to it and load an
+    #: appropriate application context.
+    AMBUDA_ENVIRONMENT = None
+
+    #: Internal secret key for encrypting sensitive data.
+    SECRET_KEY = _env("SECRET_KEY")
+
+    #: Location of the Ambuda database.
+    SQLALCHEMY_DATABASE_URI = _env("SQLALCHEMY_DATABASE_URI")
+
+    #: Where to store user uploads (PDFs, images, etc.).
+    UPLOAD_FOLDER = _env("FLASK_UPLOAD_FOLDER")
+
+    # Extensions
+    # ----------
+
+    #: If True, enable cross-site request forgery (CSRF) protection.
+    WTF_CSRF_ENABLED = True
```

Comment:
We don't use WTF_CSRF_ENABLED anywhere in the codebase? Also, why even configure this -- shouldn't this always be enabled?

Reply:
fair point -- I think an earlier draft of this PR tried centralizing all documentation in BaseConfig, but I think that's overkill. Deleted this and moved the comment to the test config.

Reply:
Oh, I didn't mean that. I think it's good to include all potential config values to BaseConfig. I just meant that I didn't see  when WTF_CSRF_ENABLED was set to False, but I do see now that it's set to False in the tests.